Sources/Authenticator/Configuration/AmplifyConfiguration.swift

```diff
@@ -35,139 +35,25 @@ struct AmplifyConfiguration {
             throw AmplifyConfigurationError.missingPlugin
         }
 
-        guard let configuration = plugin.jsonConfiguration,
-              let cognitoConfiguration = configuration.value(at: "Auth.Default") else {
-            Self.log.error("Unable to read Auth.Default from the configuration")
+        switch plugin.authConfiguration {
+        case .userPools(let configuration), .userPoolsAndIdentityPools(let configuration, _):
+            self.cognito = CognitoConfiguration(
+                usernameAttributes: configuration.usernameAttributes,
+                signupAttributes: configuration.signUpAttributes,
+                passwordProtectionSettings: configuration.passwordProtectionSettings ?? .init(minLength: 0, characterPolicy: []),
```

Comment:
❗ This is a change in existing behaviour. Currently, the Authenticator requires these password settings to be explicitly set, showing an error if that is not the case.  
But now we will just default to not having password validations.

I don't feel strongly against this, but I think Android also requires these values so we should coordinate to make the change there as well. And also call it out in the release notes

Reply:
Great callout, will follow up on this